src/script/lua_api/l_object.cpp

```diff
@@ -905,6 +905,21 @@ int ObjectRef::l_get_player_name(lua_State *L)
 	return 1;
 }
 
+// get_player_velocity(self)
+int ObjectRef::l_get_player_velocity(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	Player *player = getplayer(ref);
+	if(player == NULL){
```

Comment:
> it has to be ensured that the server actually calculates these data

Sorry, what this mean? I made a mod that sends the player its current speed, opened a server, logged in with 3 players and everything seemed to work correctly.

This is the mod:
```
function print_speed()
	local players = minetest.get_connected_players()
	for _, player in pairs(players) do
		local name = player:get_player_name()
		local velocity = player:get_player_velocity()
		minetest.chat_send_player(name, "Current Velocity: " .. minetest.pos_to_string(velocity))
	end
	minetest.after(1.0, print_speed)
end

minetest.after(1.0, print_speed)
```